web3.js/babel.config.js

```diff
@@ -0,0 +1,62 @@
+const isCalledByBundler = caller => caller.name.startsWith('@rollup');
+
+module.exports = api => {
+  const calledByBundler = api.caller(isCalledByBundler);
+  const transformEnvVariables =
+    !calledByBundler &&
+    process.env.NODE_ENV &&
+    process.env.BROWSER !== undefined;
+
+  return {
+    presets: [
+      [
+        '@babel/preset-env',
+        {
+          modules: false,
+          bugfixes: true,
+        },
+      ],
+      ['@babel/preset-typescript'],
+    ],
+    plugins: [
+      [
+        '@babel/plugin-proposal-class-properties',
+        {
+          loose: true,
+        },
+      ],
+      [
+        '@babel/plugin-proposal-private-methods',
+        {
+          loose: true,
+        },
+      ],
+      [
+        '@babel/plugin-proposal-private-property-in-object',
+        {
+          loose: true,
+        },
+      ],
+      transformEnvVariables && [
+        'transform-inline-environment-variables',
+        {
+          include: ['NODE_ENV', 'BROWSER'],
+        },
+      ],
+      transformEnvVariables && [
+        'minify-dead-code-elimination',
```

Comment:
If we consider https://github.com/solana-labs/solana/pull/21055/files#r759707513, then this becomes unnecessary as well, the consumer's bundler / minifier will remove the bit of dead code:

```js
  if (!process.env.BROWSER) {
    agentManager = new AgentManager(useHttps);
  }
```